src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,186 @@
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import classNames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import { useEffect } from "react";
+import * as Rivet from "../util/Rivet";
+import Button from "../Button/Button.jsx";
+import Icon from "../util/RivetIcons.jsx";
+import {
+  handler,
+  isRightMouseClick,
+  isTabKeyPress,
+  isUnhandledKeyPress,
+  targets,
+} from "../util/EventUtils.js";
+import { useDialog } from "@react-aria/dialog";
+import { useModalOverlay } from "../../hooks/UseModalOverlay.js";
+import { removeProperty } from "../util/RemoveProperty.js";
+
+const Dialog = ({
+  children,
+  className,
+  id = Rivet.shortuid(),
+  align,
+  closeOnOutsideClickOrEscape,
+  darkenPage,
+  disablePageInteraction,
+  isOpen,
+  onDismiss,
+  showCloseButton = true,
+  title,
+  ...attrs
+}) => {
+  const ref = React.useRef(null);
+
+  useEffect(() => {
+    handleEventRegistration();
+    return () => {
+      clickOutsideOrEscapeEventHandler.deregister();
+    };
+  }, [isOpen]);
+
+  const { dialogProps, titleProps } = useDialog(
+    { id, ["aria-labelledby"]: `${id}-title`, role: "dialog", ...attrs },
+    ref
+  );
+
+  const { modalProps, underlayProps } = useModalOverlay(
+    {
+      isDismissable: false, // using clickOutsideOrEscapeEventHandler
+      isKeyboardDismissDisabled: false, // using clickOutsideOrEscapeEventHandler
+      attrs,
+    },
+    {
+      isOpen: isOpen,
+      disablePageInteraction: disablePageInteraction,
+    },
+    ref
+  );
+
+  const handleClickOutsideOrEscape = (event) => {
+    if (event && shouldCloseDialog(event)) {
+      handleDismiss();
+      event.stopPropagation && event.stopPropagation();
+    }
+  };
+
+  const clickOutsideOrEscapeEventHandler = handler(handleClickOutsideOrEscape);
+
+  const shouldCloseDialog = (event) => {
+    // If closing is disabled, the user right clicks, presses an unhandled key, presses tab or clicks in the dialog then do not close
+    return !(
+      isRightMouseClick(event) ||
+      isUnhandledKeyPress(event) ||
+      isTabKeyPress(event) ||
+      targets(ref.current, event)
+    );
+  };
+
+  const handleEventRegistration = () => {
+    if (isOpen && closeOnOutsideClickOrEscape) {
+      clickOutsideOrEscapeEventHandler.register();
+    } else {
+      clickOutsideOrEscapeEventHandler.deregister();
+    }
+  };
+
+  const dataRvt = {
+    ...{ ...(align && { [`data-rvt-dialog-${align}`]: "" }) },
+    ...{ ...(darkenPage && { ["data-rvt-dialog-darken-page"]: "" }) },
+  };
+
+  const handleDismiss = () => {
+    onDismiss && onDismiss();
+  };
+
+  const dialogContent = (overlayProps) => (
+    <div
+      className={classNames("rvt-dialog", className)}
+      aria-hidden={!isOpen}
+      hidden={!isOpen}
+      ref={ref}
+      {...dataRvt}
+      {...overlayProps}
+    >
+      {title && (
+        <header className="rvt-dialog__header" data-testid="dialogHeader">
```

Comment:
Sounds good, I stuck to this suggestion for the Dialog related work and created https://github.com/indiana-university/rivet-react/issues/357 for other code in order to avoid merge conflicts on this PR.